src/node_file.cc

```diff
@@ -882,6 +888,62 @@ static void MKDir(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+
+static size_t CountSlashes(const char* str, size_t len) {
+  size_t cntr = 0;
+  for (size_t i = 0; i < len; i++) {
+    if (str[i] == '/') cntr++;
+  }
+  return cntr;
+}
+
+
+static int ResolveRealPath(Environment* env,
+                           std::string* ret_str,
+                           const char* path) {
+  fs_req_wrap req_wrap;
+  SYNC_CALL_NO_THROW(req_wrap, realpath, path, path);
+
+  if (err != UV_ELOOP) {
+    if (err) return err;
+    *ret_str = std::string(static_cast<const char*>(req_wrap.req.ptr));
+    return err;
+  }
+
+  std::string str_path(path);
+  size_t offset = 0;
+  size_t current = 0;
+  size_t slash_count = 0;
+
+  while ((offset = str_path.find('/', offset + 1)) != std::string::npos) {
+    if (++slash_count < 32) {
```

Comment:
From @saghul's comment in https://github.com/nodejs/node/issues/7175#issuecomment-229786187.